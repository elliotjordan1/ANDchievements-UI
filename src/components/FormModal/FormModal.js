/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { 
  Background, 
  ModalBody,
  FormTitle
} from './styles';

const FormModal = ({ title, form }) => {
  return (
    <>
    <Background >
      <ModalBody >
        <FormTitle>{title}</FormTitle>
        {form}
      </ModalBody>
    </Background>
    </>
  );
};

FormModal.propTypes = {
  title: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired
};

export default FormModal;
