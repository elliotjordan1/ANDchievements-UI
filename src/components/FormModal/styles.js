import styled from 'styled-components';
import theme from '../../global/theme';

export const FormTitle = styled.div`
  color: ${theme.colours.brand.red};
  background-color: ${theme.colours.brand.grey2};
  font: 500 20px ${theme.fontFamily.primary};
  line-height: 1.4;
  letter-spacing: 2px;
  width: 100%;
  text-indent: 8px;
  height: min-content;
  @media ${theme.media.tablet} {
    font-size: 24px;
  };
  @media ${theme.media.desktop} {
    font-size: 32px;
  };
`;

export const Background = styled.div`
  width: 100vw;
  height 100vh;
  z-index: 2;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 48px;
  @media ${theme.media.tablet}{
    top: 64px;
  };
  @media ${theme.media.desktop}{
    top: 88px;
  };
`;

export const ModalBody = styled.div`
  max-width: 800px;
  flex-direction: column;
  height: max-content;
  margin: 15vh 8px;
  z-index: 1;
  background-color: white;
  display: flex;
  @media ${theme.media.tablet}{
    margin: 30px auto;
    height: 50vh;
    max-width: 660px
  };
  @media ${theme.media.desktop}{
    margin: 30px auto;
    height: 400px;
  };
`;