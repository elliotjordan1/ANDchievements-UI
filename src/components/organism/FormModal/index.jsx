/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { 
  Background, 
  ModalBody,
  FormTitle,
  CloseIcon
} from './styles';
import Icon from '../../../assets/cross';

const FormModal = ({ title, form, close, open }) => {
  return (
    <>
    <Background open={open}>
      <ModalBody >
        <FormTitle>
          {title} 
          <CloseIcon onClick={() => {close()}} data-testid="close-icon-test"> 
            <Icon />
          </CloseIcon>
        </FormTitle>
        {form}
      </ModalBody>
    </Background>
    </>
  );
};

FormModal.propTypes = {
  title: PropTypes.string.isRequired,
  form: PropTypes.object.isRequired,
  close: PropTypes.func.isRequired,
  open: PropTypes.bool.isRequired
};

export default FormModal;
