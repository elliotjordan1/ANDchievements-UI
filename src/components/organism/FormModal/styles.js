import styled from 'styled-components';
import theme from '../../../global/theme';

export const FormTitle = styled.div`
  color: white;
  background-color: black;
  font: 500 20px ${theme.fontFamily.primary};
  line-height: 1.8;
  letter-spacing: 2px;
  width: 100%;
  text-indent: 8px;
  height: min-content;
  display: flex;
  flex-direction: row;
  @media ${theme.media.tablet} {
    font-size: 24px;
  };
  @media ${theme.media.desktop} {
    font-size: 32px;
  };
`;

export const Background = styled.div`
  width: 100vw;
  height 100vh;
  z-index: 2;
  position: fixed;
  background-color: rgba(0, 0, 0, 0.6);
  top: 48px;
  @media ${theme.media.tablet}{
    top: 64px;
  };
  @media ${theme.media.desktop}{
    top: 88px;
  };
`;

export const ModalBody = styled.div`
  max-width: 800px;
  flex-direction: column;
  height: max-content;
  margin: 15vh 8px;
  z-index: 1;
  background-color: white;
  display: flex;
  @media ${theme.media.tablet}{
    margin: 30px auto;
    max-width: 460px
  };
  @media ${theme.media.desktop}{
    margin: 30px auto;
    max-width: 730px;
  };
`;

export const CloseIcon = styled.a`
  display: flex;
  flex-direction: row;
  align-self: center;
  height: 28px;
  margin-left: 12px;
  cursor: pointer;
  & > svg{
    height: 28px;
    width: 28px;
  }
  };
`;