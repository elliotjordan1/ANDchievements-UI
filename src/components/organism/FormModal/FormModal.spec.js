import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import FormModal from '.';

describe('Form Modal component', () => {
  it('renders correctly', () => {
    const wrapper = render(<FormModal title='Test Form Title' form={<form />} close = {() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('closes correctly when closed', () => {
    const closeFunction = jest.fn();
    const { getByTestId } = render(<FormModal title='Test Form Title' form={<form />} close = {closeFunction} />);
    const closeIcon = getByTestId('close-icon-test');

    expect(closeFunction).toHaveBeenCalledTimes(0);

    fireEvent.click(closeIcon);

    expect(closeFunction).toHaveBeenCalledTimes(1);
  });
});
