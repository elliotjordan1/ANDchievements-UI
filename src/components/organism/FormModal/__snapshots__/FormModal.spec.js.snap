// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Modal component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Background-xo8fd7-1 iiyzYr"
      >
        <div
          class="styles__ModalBody-xo8fd7-2 MIEYE"
        >
          <div
            class="styles__FormTitle-xo8fd7-0 dkjxHw"
          >
            Test Form Title
            <a
              class="styles__CloseIcon-xo8fd7-3 bItPHH"
              data-testid="close-icon-test"
            >
              <svg
                fill="none"
                height="61"
                viewBox="0 0 61 61"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M48.2918 16.2921L44.7081 12.7083L30.5002 26.9162L16.2922 12.7083L12.7085 16.2921L26.9164 30.5L12.7085 44.7079L16.2922 48.2917L30.5002 34.0837L44.7081 48.2917L48.2918 44.7079L34.0839 30.5L48.2918 16.2921Z"
                  fill="white"
                />
              </svg>
            </a>
          </div>
          <form />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Background-xo8fd7-1 iiyzYr"
    >
      <div
        class="styles__ModalBody-xo8fd7-2 MIEYE"
      >
        <div
          class="styles__FormTitle-xo8fd7-0 dkjxHw"
        >
          Test Form Title
          <a
            class="styles__CloseIcon-xo8fd7-3 bItPHH"
            data-testid="close-icon-test"
          >
            <svg
              fill="none"
              height="61"
              viewBox="0 0 61 61"
              width="61"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48.2918 16.2921L44.7081 12.7083L30.5002 26.9162L16.2922 12.7083L12.7085 16.2921L26.9164 30.5L12.7085 44.7079L16.2922 48.2917L30.5002 34.0837L44.7081 48.2917L48.2918 44.7079L34.0839 30.5L48.2918 16.2921Z"
                fill="white"
              />
            </svg>
          </a>
        </div>
        <form />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
